test/unit/controllers/ChoiceControllerSpec.scala

```diff
@@ -82,41 +93,75 @@ class ChoiceControllerSpec extends ControllerSpec with ErrorHandlerMocks {
 
     "redirect to Dispatch Location page" when {
 
-      "user chooses Supplementary Dec" in new SetUp {
+      "user chooses Supplementary Dec for new declaration" in new SetUp {
         when(mockExportsCacheService.get(any())(any())).thenReturn(Future.successful(None))
         when(mockExportsCacheService.create(any[ExportsDeclaration])(any()))
-          .thenReturn(Future.successful(existingDeclaration))
+          .thenReturn(Future.successful(newDeclaration))
 
         val result = controller.submitChoice()(postRequest(supplementaryChoice))
 
         status(result) must be(SEE_OTHER)
         redirectLocation(result) must be(
           Some(controllers.declaration.routes.DispatchLocationController.displayPage().url)
         )
-        session(result).get(ExportsSessionKeys.declarationId) must be(Some("declarationId"))
+        session(result).get(ExportsSessionKeys.declarationId) must be(newDeclaration.id)
         val created: ExportsDeclaration = theCacheModelCreated
         created.id mustBe None
         created.status mustBe DeclarationStatus.DRAFT
         created.choice mustBe "SMP"
       }
 
-      "user chooses Standard Dec" in new SetUp {
+      "user chooses Supplementary Dec for existing Standard Dec" in new SetUp {
+        val existingDec = existingDeclaration(Choice.AllowedChoiceValues.StandardDec)
+        when(mockExportsCacheService.get(any())(any())).thenReturn(Future.successful(Some(existingDec)))
+        when(mockExportsCacheService.update(any())(any()))
+          .thenReturn(Future.successful(Some(existingDec)))
```

Comment:
Those mocks are in `MockExportCacheService`

`withNewCaching(existingDec)`

Reply:
Thanks.  I'll take a look

Reply:
Just had a look.  `withNewCaching` mocks all three of the get/create/update methods.  I specifically want to test that update (not create) is called when dealing with an existing draft dec (and vise-versa for a new one).    "Blanket" mocking all three methods might have hidden a bug.  (Accept that I could use verify to check what mocked methods are actually called but a null-pointer tends to grab your attention)

Reply:
In fact, there are quite a few tests here that are needlessly mocking `mockExportsCacheService.get` (i.e. tests pass without mocking that method - as they should)